src/auto_note.cpp

```diff
@@ -0,0 +1,454 @@
+        if( !extra.autonote ) {
+            continue;
+        }
+
+        // Check if auto note is enabled for this map extra type
+        bool isAutoNoteEnabled = settings.has_auto_note_enabled( extra.id );
+
+        // Insert into cache
+        this->mapExtraCache.emplace( std::make_pair( extra.id.str(), std::make_pair( extra,
+                                     isAutoNoteEnabled ) ) );
+
+        // Only insert into display cache if map extra was already discovered. This
+        // is done in order to avoid spoilers
+        if( settings.was_discovered( extra.id ) ) {
+            this->displayCache.push_back( extra.id.str() );
+        }
+    }
+}
+
+auto_note_manager_gui::auto_note_manager_gui()
+{
+    // Initialize GUI state
+    this->initialize();
+}
+
+bool auto_note_manager_gui::was_changed()
+{
+    return this->wasChanged;
+}
+
+void auto_note_manager_gui::show()
+{
+    const int iHeaderHeight = 3;
+    const int iContentHeight = FULL_SCREEN_HEIGHT - 2 - iHeaderHeight;
+
+    const int iOffsetX = TERMX > FULL_SCREEN_WIDTH ? ( TERMX - FULL_SCREEN_WIDTH ) / 2 : 0;
+    const int iOffsetY = TERMY > FULL_SCREEN_HEIGHT ? ( TERMY - FULL_SCREEN_HEIGHT ) / 2 : 0;
+
+    catacurses::window w_help = catacurses::newwin( FULL_SCREEN_HEIGHT / 2 + 2,
+                                FULL_SCREEN_WIDTH * 3 / 4,
+                                point( iOffsetX + 19 / 2, 7 + iOffsetY + FULL_SCREEN_HEIGHT / 2 / 2 ) );
+
+    catacurses::window w_border = catacurses::newwin( FULL_SCREEN_HEIGHT, FULL_SCREEN_WIDTH,
+                                  point( iOffsetX, iOffsetY ) );
+
+    catacurses::window w_header = catacurses::newwin( iHeaderHeight, FULL_SCREEN_WIDTH - 2,
+                                  point( 1 + iOffsetX, 1 + iOffsetY ) );
+
+    catacurses::window w = catacurses::newwin( iContentHeight, FULL_SCREEN_WIDTH - 2,
+                           point( 1 + iOffsetX, iHeaderHeight + 1 + iOffsetY ) );
+
+    // ===========================================================================
+    // Perform initial draw. This includes things like the window border that do
+    // not need to be refreshed more than once.
+
+    // == Draw border
+    draw_border( w_border, BORDER_COLOR, _( " AUTO NOTES MANAGER " ) );
+    mvwputch( w_border, point( 0, 2 ), c_light_gray, LINE_XXXO );
+    mvwputch( w_border, point( 79, 2 ), c_light_gray, LINE_XOXX );
+    mvwputch( w_border, point( 61, FULL_SCREEN_HEIGHT - 1 ), c_light_gray, LINE_XXOX );
+    wrefresh( w_border );
+
+    // == Draw header
+    int tmpx = 0;
+    tmpx += shortcut_print( w_header, point( tmpx, 0 ), c_white, c_light_green, _( "<E>nable" ) ) + 2;
+    tmpx += shortcut_print( w_header, point( tmpx, 0 ), c_white, c_light_green, _( "<D>isable" ) ) + 2;
+    tmpx += shortcut_print( w_header, point( tmpx, 0 ), c_white, c_light_green,
+                            _( "<Enter>-Toggle" ) ) + 2;
+    ( void )tmpx;
```

Comment:
What's the purpose of this statement?

Reply:
`clang-tidy` CI fails if the result in `tmpx` after the compound statement is not used, and also if the result of the last `shortcut_print` is ignored. If theres a different solution I'm all ears.

Reply:
Add a comment before the last line probably. #30418
```cpp
// NOLINTNEXTLINE(clang-analyzer-deadcode.DeadStores)
```

Reply:
Thanks, I applied that change. Lets see if it works with the CI.